apps/src/code-studio/hiddenStageRedux.js

```diff
@@ -5,57 +5,64 @@
 import $ from 'jquery';
 import { INIT_PROGRESS } from './progressRedux';
 import experiments from '@cdo/apps/experiments';
+import Immutable from 'immutable';
 
-export const UPDATE_HIDDEN_STAGES = 'hiddenStage/UPDATE_HIDDEN_STAGES';
+export const UPDATE_HIDDEN_STAGE = 'hiddenStage/UPDATE_HIDDEN_STAGE';
 export const ALLOW_HIDEABLE = 'hiddenStage/ALLOW_HIDEABLE';
 
+const STUDENT_SECTION_ID = 'STUDENT';
+
 export const hiddenStagesEnabled = () => experiments.isEnabled('hiddenStages');
 
-// State is a key indicating whether or not we've received hidden stage info
-// from the server, and then a set of stageIds mapped to bools that are true
-// if that stage is hidden
-const initialState = {
-  initialized: false
-};
+const initialState = Immutable.fromJS({
+  initialized: false,
+  // mapping of section id to hidden stages for that section
+  // Teachers will potentially have a number of section ids. For students we
+  // use a sectionId of STUDENT_SECTION_ID, which represents the hidden state
+  // for the student based on the sections they are in.
```

Comment:
This is a little bit strange, in that I have our API provide differently shaped values depending on if you're a teacher or student. If a teacher, we provide a map from a sectionId to a list of stageIds. If a student, we just provide a list of stageIds. In the latter case, we then turn that into a map on the client where the section id is STUDENT_SECTION_ID. Not sure I can come up with a cleaner way to do all this.